app/models/hostgroup.rb

```diff
@@ -212,6 +213,12 @@ def update_ancestry_puppetclasses
     unscoped_find(ancestry.to_s.split('/').last.to_i).update_puppetclasses_total_hosts if ancestry.present?
   end
 
+  def children_hosts_count
+    count = hosts_count
+    return count unless children.any?
+    count += children.map(&:children_hosts_count).reduce(:+)
```

Comment:
yeah, I know. for some reason it does not return the correct values so left it this way